codex/conf.nim

```diff
@@ -292,9 +293,17 @@ proc defaultDataDir*(): string =
 
   getHomeDir() / dataDir
 
-proc parseCmdArg*(T: type MultiAddress, input: string): T
+proc parseCmdArg*(T: typedesc[MultiAddress],
+                  input: string): MultiAddress
                  {.upraises: [ValueError, LPError].} =
-  MultiAddress.init($input).tryGet()
+  var ma: MultiAddress
```

Comment:
should probably use `without ma =? ...:`

Reply:
Looks like without is useable here.

Reply:
@dryajov, changing this to use `without` causes a compilation failure on Windows. It's something I've run into a few times and only occurs on Windows (?). 

```nim
D:\a\nim-codex\nim-codex\codex\conf.nim(299, 59) Error: Invalid node kind nnkStmtList for macros.`$`
```

Here's the error log: https://github.com/codex-storage/nim-codex/actions/runs/5720835236/job/15501441283#step:8:386

Reply:
I filed an issue for this in questionable since it's popped up randomly a couple of times: https://github.com/codex-storage/questionable/issues/39